server/grpc_asset.go

```diff
@@ -144,28 +125,25 @@ func (s *grpcServer) fetchItem(uri string, expectedHash string) (bool, string, i
 		return false, "", int64(-1)
 	}
 
-	_, simple := simpleExtensions[filepath.Ext(u.Path)]
-	if !simple {
-		return false, "", int64(-1)
-	}
-
```

Comment:
If whitelisting won't work, I wonder if we should blacklist URLs that we can safely assume won't work, like *.git ?